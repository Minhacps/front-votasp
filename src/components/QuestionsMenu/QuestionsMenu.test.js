import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import QuestionMenu from './QuestionsMenu';
import { answersMock, questionsMock } from './QuestionsMenuMock';

describe('QuestionsMenu', () => {
  it('should match a snapshot', () => {
    const questionMenu = shallow((
      <QuestionMenu
        answersArray={answersMock}
        questionsArray={questionsMock}
        currentQuestion={30}
      />
    ))

    expect(toJson(questionMenu)).toMatchSnapshot();
  })

  it('should match a snapshot when menu is open', () => {
    const questionMenu = shallow((
      <QuestionMenu
        answersArray={answersMock}
        questionsArray={questionsMock}
        userAnswers={[]}
        currentQuestion={30}
      />
    ))

    questionMenu.find('.questions-button').simulate('click');
    expect(toJson(questionMenu)).toMatchSnapshot();
  })

  it('should toggle questions board when click on questions button', () => {
    const questionMenu = shallow((
      <QuestionMenu
        answersArray={answersMock}
        questionsArray={questionsMock}
        userAnswers={[]}
        currentQuestion={30}
      />
    ))

    questionMenu.find('.questions-button').simulate('click');
    expect(questionMenu.find('.questions-board')).toHaveLength(1)

    questionMenu.find('.questions-button').simulate('click');
    expect(questionMenu.find('.questions-board')).toHaveLength(0)

  })
})
