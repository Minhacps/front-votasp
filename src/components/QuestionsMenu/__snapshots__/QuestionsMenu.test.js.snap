// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionsMenu should match a snapshot 1`] = `
<div
  className="questions-menu"
>
  <button
    className="questions-button"
    onClick={[Function]}
    type="button"
  >
    Questões
    <svg
      className="arrow"
      height="15"
      viewBox="0 0 24 24"
      width="15"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 16.67l2.829 2.83 9.175-9.339 9.167 9.339 2.829-2.83-11.996-12.17z"
      />
    </svg>
  </button>
</div>
`;

exports[`QuestionsMenu should match a snapshot when menu is open 1`] = `
<div
  className="questions-menu active"
>
  <button
    className="questions-button"
    onClick={[Function]}
    type="button"
  >
    Questões
    <svg
      className="arrow"
      height="15"
      viewBox="0 0 24 24"
      width="15"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 16.67l2.829 2.83 9.175-9.339 9.167 9.339 2.829-2.83-11.996-12.17z"
      />
    </svg>
  </button>
  <div
    className="questions-board"
  >
    <ol>
      <li
        className="question-item"
        key="1"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
      </li>
      <li
        className="question-item"
        key="2"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
      </li>
      <li
        className="question-item"
        key="3"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
      </li>
      <li
        className="question-item"
        key="4"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
      </li>
      <li
        className="question-item"
        key="5"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
      </li>
      <li
        className="question-item"
        key="6"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
      </li>
      <li
        className="question-item"
        key="7"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
      </li>
      <li
        className="question-item"
        key="8"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
      </li>
      <li
        className="question-item"
        key="9"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
      </li>
      <li
        className="question-item"
        key="10"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          10
        </button>
      </li>
      <li
        className="question-item"
        key="11"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          11
        </button>
      </li>
      <li
        className="question-item"
        key="12"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          12
        </button>
      </li>
      <li
        className="question-item"
        key="13"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          13
        </button>
      </li>
      <li
        className="question-item"
        key="14"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          14
        </button>
      </li>
      <li
        className="question-item"
        key="15"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          15
        </button>
      </li>
      <li
        className="question-item"
        key="16"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          16
        </button>
      </li>
      <li
        className="question-item"
        key="17"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          17
        </button>
      </li>
      <li
        className="question-item"
        key="18"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          18
        </button>
      </li>
      <li
        className="question-item"
        key="19"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          19
        </button>
      </li>
      <li
        className="question-item"
        key="20"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          20
        </button>
      </li>
      <li
        className="question-item"
        key="21"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          21
        </button>
      </li>
      <li
        className="question-item"
        key="22"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          22
        </button>
      </li>
      <li
        className="question-item"
        key="23"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          23
        </button>
      </li>
      <li
        className="question-item"
        key="24"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          24
        </button>
      </li>
      <li
        className="question-item"
        key="25"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          25
        </button>
      </li>
      <li
        className="question-item"
        key="26"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          26
        </button>
      </li>
      <li
        className="question-item"
        key="27"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          27
        </button>
      </li>
      <li
        className="question-item"
        key="28"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          28
        </button>
      </li>
      <li
        className="question-item"
        key="29"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          29
        </button>
      </li>
      <li
        className="question-item"
        key="30"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          30
        </button>
      </li>
      <li
        className="question-item current-question"
        key="31"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          31
        </button>
      </li>
      <li
        className="question-item"
        key="32"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          32
        </button>
      </li>
      <li
        className="question-item"
        key="33"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          33
        </button>
      </li>
      <li
        className="question-item"
        key="34"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          34
        </button>
      </li>
      <li
        className="question-item"
        key="35"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          35
        </button>
      </li>
      <li
        className="question-item"
        key="36"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          36
        </button>
      </li>
      <li
        className="question-item"
        key="37"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          37
        </button>
      </li>
      <li
        className="question-item"
        key="38"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          38
        </button>
      </li>
      <li
        className="question-item"
        key="39"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          39
        </button>
      </li>
      <li
        className="question-item"
        key="40"
      >
        <button
          className="question-item-button"
          onClick={[Function]}
          type="button"
        >
          40
        </button>
      </li>
      <li
        className="current-question"
      />
    </ol>
  </div>
</div>
`;
